storage/database_test.go

```diff
@@ -0,0 +1,471 @@
+}
+
+// TestBadger_Default_WriteBatch_Experiment_10MB_Batch is a BadgerDB test in which we write 10MB of data to the database.
+// In the test, we perform identical operations as in TestBadger_Default_Experiment_10MB_Batch. This experiment uses the
+// "Default" Badger config, and Badger's WriteBatch.
+func TestBadger_Default_WriteBatch_Experiment_10MB_Batch(t *testing.T) {
+	require := require.New(t)
+
+	testConfig := &TestConfig{
+		BatchSizeBytes:      10000000,
+		BatchSizeItems:      100,
+		BatchItemsRemoved:   20,
+		BatchItemsRetrieved: 20,
+		BatchItemsIterated:  20,
+	}
+	dir, err := os.MkdirTemp("", "badgerdb-default-writebatch-10mb")
+	t.Logf("BadgerDB directory: %s\nIt should be automatically removed at the end of the test", dir)
+	require.NoError(err)
+
+	opts := DefaultBadgerOptions(dir)
+	db := NewBadgerDatabase(opts, true)
+	require.NoError(db.Setup())
+	defer db.Erase()
+	defer db.Close()
+
+	badgerCtx := db.GetContext([]byte{})
+	GenericTest(db, badgerCtx, testConfig, t)
+}
+
+// TestBadger_Default_WriteBatch_Experiment_25MB_Batch is a BadgerDB test in which we write 25MB of data to the database.
+// In the test, we perform identical operations as in TestBadger_Default_Experiment_10MB_Batch. This experiment uses the
+// "Default" Badger config, and Badger's WriteBatch.
+func TestBadger_Default_WriteBatch_Experiment_25MB_Batch(t *testing.T) {
+	require := require.New(t)
+
+	testConfig := &TestConfig{
+		BatchSizeBytes:      25000000,
+		BatchSizeItems:      250,
+		BatchItemsRemoved:   20,
+		BatchItemsRetrieved: 20,
+		BatchItemsIterated:  20,
+	}
+	dir, err := os.MkdirTemp("", "badgerdb-default-writebatch-25mb")
+	t.Logf("BadgerDB directory: %s\nIt should be automatically removed at the end of the test", dir)
+	require.NoError(err)
+
+	opts := DefaultBadgerOptions(dir)
+	db := NewBadgerDatabase(opts, true)
+	require.NoError(db.Setup())
+	defer db.Erase()
+	defer db.Close()
+
+	badgerCtx := db.GetContext([]byte{})
+	GenericTest(db, badgerCtx, testConfig, t)
+}
+
+// TestBadger_Default_WriteBatch_Experiment_100MB_Batch is a BadgerDB test in which we write 100MB of data to the database.
+// In the test, we perform identical operations as in TestBadger_Default_Experiment_10MB_Batch. This experiment uses the
+// "Default" Badger config, and Badger's WriteBatch.
+func TestBadger_Default_WriteBatch_Experiment_100MB_Batch(t *testing.T) {
+	require := require.New(t)
+
+	testConfig := &TestConfig{
+		BatchSizeBytes:      100000000,
+		BatchSizeItems:      1000,
+		BatchItemsRemoved:   20,
+		BatchItemsRetrieved: 20,
+		BatchItemsIterated:  20,
+	}
+	dir, err := os.MkdirTemp("", "badgerdb-default-writebatch-100mb")
+	t.Logf("BadgerDB directory: %s\nIt should be automatically removed at the end of the test", dir)
+	require.NoError(err)
+
+	opts := DefaultBadgerOptions(dir)
+	db := NewBadgerDatabase(opts, true)
+	require.NoError(db.Setup())
+	defer db.Erase()
+	defer db.Close()
+
+	badgerCtx := db.GetContext([]byte{})
```

Comment:
It seems like a lot of these tests do exactly the same thing. can we refactor a bit so it's easier to read? something like:
```
func _doBadgerTest(testConfig *TestConfig, dir string, useDefaultOptions bool)
```

or just pass badger options in. This way we don't have so much clutter. 

Also, do we really need this many tests for opening performance badger DBs and writing to disk? Are we going to run these tests in CI? If we're running them in CI, I would recommend we reduce the number of tests to what is absolutely necessary so we're not clogging up CI runs. Additionally, limiting the number of usages of performance options (when running in CI) would be great.

Reply:
Agreed, I think just one generic test is sufficient to stress test the badger db implementation. So I got rid of all of them but one.